diesel_tests/tests/transactions.rs

```diff
@@ -55,6 +55,34 @@ fn transaction_is_rolled_back_when_returned_an_error() {
     drop_test_table(connection, test_name);
 }
 
+#[test]
+#[cfg(feature = "sqlite")]
```

Comment:
```suggestion
```

Should we run this test on both mysql and pg?

Reply:
Based on the [CI result](https://github.com/diesel-rs/diesel/actions/runs/953354716) of the first commit, sqlite and mysql are good. `pg` failed the test. 

- for sqlite, I am confident that the test works as expected because of [sqlite3 doc](https://www.sqlite.org/lang_transaction.html) (see the section near the bottom).

- for mysql, I didn't find clear doc about rollback behavior, but the CI result shows the test works.

- for pg, I think the test didn't work because `pg` probably not treat it as error when a transaction is already rollback, based on my understanding of [their doc](https://www.postgresql.org/docs/9.4/sql-rollback.html). 

If we know for sure that mysql ROLLBACK command behaves the same as sqlite3, in the regard of executing 2nd time would trigger an error, then I can add mysql into the feature list here, which will be better than only having sqlite.  Or if there is another (better) way to cause a rollback fail for all of three backends,  I will certainly adopt.

Reply:
I've added more comments to describe the rationale for the test.

Reply:
Seems reasonable for me :+1: